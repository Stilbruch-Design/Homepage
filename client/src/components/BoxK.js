import React, { Component } from "react";
import { connect } from "react-redux";
import FormC from "./FormC";
import FormNotification from "./FormNotification";

class BoxK extends Component {
  render() {
    return (
      <div className="myFlex navChange" id="Kontakt">
        <div className="flexHeader">Kontaktform</div>
        <div className="flexMainItems">
          {this.props.form.formSend ? <FormNotification /> : <FormC />}
        </div>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  form: state.form
});

export default connect(
  mapStateToProps,
  {}
)(BoxK);
