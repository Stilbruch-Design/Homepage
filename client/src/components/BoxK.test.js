import React from "react";
import { createStore, applyMiddleware } from "redux";
import thunk from "redux-thunk";
import { Provider } from "react-redux";
import {
  render,
  fireEvent,
  cleanup,
  waitForElement
} from "@testing-library/react";
import rootReducer from "../reducers";
import "jest-dom/extend-expect";
import userEvent from "@testing-library/user-event";
import BoxK from "./BoxK";

afterEach(cleanup);

const middleware = [thunk];

function renderWithRedux(
  ui,
  {
    initialState,
    store = createStore(
      rootReducer,
      initialState,
      applyMiddleware(...middleware)
    )
  } = {}
) {
  return {
    ...render(<Provider store={store}>{ui}</Provider>),
    store
  };
}

test("<BoxK /> shows FormNotification when all form fields are valid", async () => {
  const { getByTestId, queryByTestId, debug } = renderWithRedux(<BoxK />);

  userEvent.type(getByTestId("InputFirstName"), "Michael");
  userEvent.type(getByTestId("InputLastName"), "HÃ¼bner");
  userEvent.type(getByTestId("InputEmail"), "m.huebner@email.com");
  userEvent.type(getByTestId("TextareaMessage"), "This is a test!");

  expect(getByTestId("InputEmail")).toHaveAttribute(
    "value",
    "m.huebner@email.com"
  );

  expect(getByTestId("TextareaMessage").value).toBe("This is a test!");

  debug();

  fireEvent.click(getByTestId("ButtonSubmit"));

  debug();

  expect(queryByTestId("FirstName_Error")).not.toBeInTheDocument();
  expect(queryByTestId("LastName_Error")).not.toBeInTheDocument();
  expect(queryByTestId("Email_Error")).not.toBeInTheDocument();
  expect(queryByTestId("Message_Error")).not.toBeInTheDocument();

  debug();

  const FormSend = await waitForElement(() => getByTestId("FormSend"));

  expect(FormSend).toBe(true);
});
