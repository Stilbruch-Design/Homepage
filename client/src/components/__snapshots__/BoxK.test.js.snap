// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BoxK /> renders 1`] = `
<div
  class="navChange sc-EHOje dOAomh"
  id="Kontakt"
>
  <div
    class="sc-bZQynM hRZpia"
  >
    Kontaktform
  </div>
  <div
    class="sc-gzVnrw ftxNwV"
  >
    <form
      action="/kontakt"
      class="navChange"
      data-testid="FormMain"
      id="kontakt"
      method="POST"
    >
      <input
        class="sc-bwzfXH eAykMC"
        name="firstName1"
        style="display: none;"
        type="text"
        value=""
      />
      <input
        class="sc-bwzfXH eAykMC"
        name="lastName1"
        style="display: none;"
        type="text"
        value=""
      />
      <input
        class="sc-bwzfXH eAykMC"
        data-testid="InputEmail1"
        name="email1"
        style="display: none;"
        type="email"
        value=""
      />
      <textarea
        class="sc-htpNat kDgGqi"
        name="message1"
        style="display: none;"
      />
      <input
        class="sc-bwzfXH eAykMC"
        data-testid="InputFirstName"
        minlength="2"
        name="firstName"
        placeholder="Vorname"
        required=""
        type="text"
        value=""
      />
      <input
        class="sc-bwzfXH eAykMC"
        data-testid="InputLastName"
        minlength="2"
        name="lastName"
        placeholder="Nachname"
        required=""
        type="text"
        value=""
      />
      <input
        class="sc-bwzfXH eAykMC"
        data-testid="InputEmail"
        name="email"
        placeholder="Email"
        required=""
        type="email"
        value=""
      />
      <textarea
        class="sc-htpNat kDgGqi"
        data-testid="TextareaMessage"
        name="message"
        placeholder="Bitte schreibe hier deine Nachricht."
        required=""
      />
      <button
        class="sc-bdVaJa kIdCTB"
        data-testid="ButtonSubmit"
        type="submit"
      >
        <span>
          Nachricht senden
        </span>
      </button>
    </form>
  </div>
</div>
`;
